Neos.Flow/Classes/Persistence/Doctrine/Mapping/Driver/FlowAnnotationDriver.php

```diff
@@ -548,11 +553,11 @@ protected function buildJoinColumnsIfNeeded(array $joinColumns, array $mapping,
         foreach ($joinColumns as &$joinColumn) {
             if ($joinColumn['referencedColumnName'] === null || $joinColumn['referencedColumnName'] === 'id') {
                 if ($direction === self::MAPPING_REGULAR) {
-                    $idProperties = $this->reflectionService->getPropertyNamesByTag($mapping['targetEntity'], 'id');
+                    $idProperties = $this->reflectionService->getPropertyNamesByAnnotation($mapping['targetEntity'], ORM\Id::class);
                     $joinColumnName = $this->buildJoinTableColumnName($mapping['targetEntity']);
                 } else {
                     $className = $this->getUnproxiedClassName($property->getDeclaringClass()->getName());
-                    $idProperties = $this->reflectionService->getPropertyNamesByTag($className, 'id');
+                    $idProperties = $this->reflectionService->getPropertyNamesByAnnotation($className, ORM\Id::class);
```

Comment:
I think this is a bug in the current implementation, because annotations like `@Foo\Id` are counted as id property too. Switching to `ORM\Id` is more safe and I think also the correct behaviour.